src/Components/Endpoints/src/FormMapping/FormDataMapperOptions.cs

```diff
@@ -25,6 +26,21 @@ public FormDataMapperOptions()
         _factories.Add(new ComplexTypeConverterFactory(this));
     }
 
+    [RequiresDynamicCode(FormMappingHelpers.RequiresDynamicCodeMessage)]
+    [RequiresUnreferencedCode(FormMappingHelpers.RequiresUnreferencedCodeMessage)]
+    internal FormDataMapperOptions(IHttpContextAccessor httpContextAccessor)
+    {
+        // We don't use the base constructor here since the ordering of the factories is important.
+        _converters = new(WellKnownConverters.Converters);
+        _factories.Add(new ParsableConverterFactory());
+        _factories.Add(new EnumConverterFactory());
+        _factories.Add(new FileConverterFactory(httpContextAccessor));
```

Comment:
Could we expose the file collection inside `FormDataReader`? That would get rid of the need for the httpContextAccessor

Reply:
> Could we expose the file collection inside FormDataReader? 

We could! I avoided it because I wasn't sure if we wanted to make the `FormDataReader` HTTP-specific, but it's definitely my preferred option. I can experiment with this approach.